File: Problems/1422. Maximum Score After Splitting a String/1422. Maximum Score After Splitting a String.java
class Solution {
    public int maxScore(String s) {
        int l_comp = (s.charAt(0) == '0') ? 1 : 0;
        int best_l_comp = l_comp;
        int n_ones = 0;
        for (int i = 1; i < s.length()-1; i++) {
            if (s.charAt(i) == '0') l_comp++;
            else {
                l_comp--;
                n_ones++;
            }

            if (l_comp > best_l_comp) best_l_comp = l_comp;
        }
        n_ones += (s.charAt(s.length()-1) == '1') ? 1 : 0;

        return best_l_comp + n_ones;
    }
}

File: Problems/1422. Maximum Score After Splitting a String/1422. Maximum Score After Splitting a String.py
class Solution:
    def maxScore(self, s: str) -> int:
        l_comp = 1 if s[0] == '0' else 0
        best_l_comp = l_comp
        num_ones = 0
        for bit in s[1:-1]:
            if bit == '0':
                l_comp += 1
                best_l_comp = max(l_comp, best_l_comp)
            else:
                l_comp -= 1
                num_ones += 1
        num_ones += 1 if s[-1] == '1' else 0
        
        return best_l_comp + num_ones